### PR TITLE
Redo the build system

### PR DESCRIPTION
**Pull request by nicolasvasilache**
This commit is the first step towards redoing the build system.
Given the recent developments in PyTorch world and the complexity now
involved in building ATen, it has become counter-productive to build
everything from scratch.

Instead, we use conda to get the dependencies we need with the exception
of:
1. ISL, we build that from source and link it directly in our CMakeLists.txt
2. Halide, that we still build from source until a more official packaging
becomes available
3. Tapir, punting on conda packaging this for now
4. Caffe2, deactivated for now but that will be reenabled soon and squashed
here

The other dependencies should come from conda (i.e. glog, gflags, gtest,
protobuf, ATen, ...)

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Prerequisite
+```
+export CLANG_PREFIX=$($HOME/clang+llvm-tapir5.0/bin/llvm-config --prefix);
+export LLVM_SOURCES-/tmp/llvm_sources-tapir5.0;
```
Comment (ftynse) [inline BUILD.md]: typo here, `=` probably intended

File: BUILD.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Prerequisite
+```
+export CLANG_PREFIX=$($HOME/clang+llvm-tapir5.0/bin/llvm-config --prefix);
```
Comment (ftynse) [inline BUILD.md]: I wouldn't hardcode LLVM install locations

File: BUILD.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+conda install -y gflags glog gtest protobuf
+conda install -y pyyaml mkl-include
+conda install -yc conda-forge pytest
+conda install pytorch torchvision cuda90 -c pytorch
+```
+
+# Activate preinstalled conda in your current terminal
+```
+. ~/conda/miniconda/bin/activate
+condo activate tc_build
```
Comment (ftynse) [inline BUILD.md]: typo: `conda`

File: BUILD.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+```
+
+# Build TC
+The following assumes a predefined location for `CUDA_TOOLKIT_ROOT_DIR`, `CUDNN_ROOT_DIR` and `CLANG_PREFIX`, adapt to your needs.
+```
+CUDA_TOOLKIT_ROOT_DIR=/public/apps/cuda/9.0/ CUDNN_ROOT_DIR=/public/apps/cudnn/v7.0/cuda/ CLANG_PREFIX=$($HOME/clang+llvm-tapir5.0/bin/llvm-config --prefix) ./build.sh --all
```
Comment (ftynse) [inline BUILD.md]: Shouldn't `CUDA_TOOLKIT_ROOT_DIR` come from conda?  We have installed cuda-9.0 with it.

Comment (nicolasvasilache) [reply BUILD.md]: We have not installed cuda-9.0 with conda, it's just an option for installing pytorch.
That being said it does indeed pull some dependencies but only libraries (no headers and no binaries).

**Comment by skimo-openhub** [comment]
I assume this has been superseded by #451.

**Pull request closed without merging**